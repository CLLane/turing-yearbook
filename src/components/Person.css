
.person {
  display: flex;
  flex-direction: column;
  height: 450px;
  width: 80%;
  padding: 10px;
  background-color: rgba(86, 109, 185, 0.90);
  border-radius: 10px;
  justify-content: center;
  align-items: center;
}

img {
  height: 50%;
  width: 75%;
  border-radius: 10px;
  border: solid 3px darkblue
}

p {
  font-weight: 700;
}

span {
  font-weight: bold;
  color: blue;
}

.button {
  width: 75%;
  height: 25px;
  border-radius: 10px;
}