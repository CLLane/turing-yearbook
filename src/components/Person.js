import React, { Component } from 'react';
import './Person.css';

class Person extends Component {
  constructor(props){
    super(props)
    this.state = {
      id: this.props.id,
      name: this.props.name,
      quote: this.props.quote,
      superlative: this.props.superlative,
      photo: this.props.photo
    }
  }

  editHandler(event) {
    this.setState({ [event.target.id]: event.target.innerText })
  }

  render() {
    return (
      <div className='person'data-id={this.props.id}>
        <h3 
          contentEditable='true'
          suppressContentEditableWarning={true}
          onKeyUp={event => this.editHandler(event, 'name')}
          onBlur={this.editSave}
          >{this.state.name}
        </h3>
        <img src={this.state.photo} alt="the essence of one\'s being"></img>
        <p
          contentEditable='true'
          suppressContentEditableWarning={true}
          onKeyUp={event => this.editHandler(event, 'quote')}
          onBlur={this.editSave}
          >{this.state.quote}
        </p>
        <p 
           contentEditable='true'
           suppressContentEditableWarning={true}
           onKeyUp={event => this.editHandler(event, 'superlative')}
           onBlur={this.editSave}
           > {this.state.superlative}
        </p>
        <button className='button' onClick={() => this.props.deleteStudent(this.state.id)}>Delete</button>
      </div>
    )
  }
}
export default Person;
