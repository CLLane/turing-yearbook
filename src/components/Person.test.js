import React from 'react';
import { shallow } from 'enzyme';
import Person from './Person';

describe('Person', () => {
  it('should render to match the snapshot', () => {
    const wrapper = shallow(<Person 
      id={9999}
      key={9999}
      name='Chris'
      quote='Greg is awesome'
      superlative='Mostlikely to ask Greg for testing help'
      photo='cheese.jpeg'
      deleteStudent= {jest.fn()}
    />)

    expect(wrapper).toMatchSnapshot()
  });
});