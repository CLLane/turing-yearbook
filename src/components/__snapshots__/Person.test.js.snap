// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should render to match the snapshot 1`] = `
<div
  className="person"
  data-id={9999}
>
  <h3
    contentEditable="true"
    onKeyUp={[Function]}
    suppressContentEditableWarning={true}
  >
    Chris
  </h3>
  <img
    alt="the essence of one\\\\'s being"
    src="cheese.jpeg"
  />
  <p
    contentEditable="true"
    onKeyUp={[Function]}
    suppressContentEditableWarning={true}
  >
    Greg is awesome
  </p>
  <p
    contentEditable="true"
    onKeyUp={[Function]}
    suppressContentEditableWarning={true}
  >
     
    Mostlikely to ask Greg for testing help
  </p>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
